File: app/src/main/java/com/example/quiz/ui/SplashActivity.java
package com.example.quiz.ui;

import android.content.Intent;
import android.graphics.drawable.AnimationDrawable;
import android.os.Bundle;
import android.os.CountDownTimer;
import androidx.appcompat.app.AppCompatActivity;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.example.quiz.R;
import com.example.quiz.ScreenActivity;
import com.example.quiz.game.MathGameActivity;
import com.romainpiel.shimmer.Shimmer;
import com.romainpiel.shimmer.ShimmerTextView;


public class SplashActivity extends AppCompatActivity {
    ImageView imageView;
    Shimmer shimmer;
    ShimmerTextView myShimmerTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);

        myShimmerTextView = findViewById(R.id.myShimmerTextView);
        imageView = findViewById(R.id.imageView);

        Glide.with(this)
                .asGif()
                .load(R.drawable.coding)
                .into(imageView);

        new CountDownTimer(7000,1000){
            public void onTick(long millisUntilFinished){
                shimmer = new Shimmer();
                shimmer.start(myShimmerTextView);
            }
            public void onFinish(){
                Intent intent = new Intent(SplashActivity.this, MathGameActivity.class);
                startActivity(intent);
                finish();
            }
        } .start();

    }

}

File: app/src/main/java/com/example/quiz/ui/SplashActivity.kt
package com.example.quiz.ui

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.os.CountDownTimer
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import com.example.quiz.R
import com.example.quiz.ScreenActivity
import com.example.quiz.databinding.ActivitySplashScreenBinding
import com.romainpiel.shimmer.Shimmer

@SuppressLint("CustomSplashScreen")
class SplashActivity : AppCompatActivity() {
    var shimmer: Shimmer? = null
    private lateinit var binding: ActivitySplashScreenBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySplashScreenBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        binding.imageView.let {
            Glide.with(this)
                .asGif()
                .load(R.drawable.coding)
                .into(it)
        }
        object : CountDownTimer(7000, 1000) {
            override fun onTick(millisUntilFinished: Long) {
                shimmer = Shimmer()
                shimmer!!.start(binding.myShimmerTextView)
            }

            override fun onFinish() {
                val intent = Intent(this@SplashActivity, ScreenActivity::class.java)
                startActivity(intent)
                finish()
            }
        }.start()
    }
}